OpenSourcePolitics/decidim-module-ptp
# Bump/0.29 budgets booth

####🎩 Description

This PR updates decidim-budgets_booth to 0.29 version, and also deletes old functionalities and updates others (see Readme file for details).


####Testing
1. To test "Defining cancel voting redirection destinations"
- as an admin, go to a process, modify the budget's component by adding an "URL to redirect user when canceling voting" like "http://localhost:3000/processes/".
- in the application, go to the budgets of the process modified above. If there is several budgets, click on vote in one of them.
- on the projects index page, click on "add to vote" on one project, and then click on 'Exit voting booth', confirm 'Exit voting', and see that you are redirected to processes.

2. To test "Defining after finishing voting redirection"
- as an admin, go to a process, modify the budget's component by adding an "URL to redirect user after voting on all available budgets" like "http://localhost:3000/processes/".
- in the Global Settings workflow, choose "vote in one"
- in the application, go to the budgets of the process modified above. If there is several budgets, click on vote in one of them.
- on the projects index page, click on "add to vote" on one project, if it is already possible, click on "Vote budget", otherwise add another project to the vote. Confirm your vote and see that you are redirected to processes.

3. To test Adding after voting message and after completing voting message
- as an admin, go to a process with only 2 budgets, modify the budget's component by adding a custom text n°1 in "Popup text after each vote", and add a custom text n°2 in "Popup text after voting in all available budgets"
- in the Global Settings workflow, choose "vote in all", and check that "Maximum number of budgets that user can vote on" is set to 0.
- in the application, go to the budgets of the process modified above. Choose one budget,  complete the voting process for it and see that you have a modal with you custom message n°1
- complete the voting process for the second budget, and see that you have a modal with custom message n°2 as you have voted for all available budgets.

4. To test for maximum number of budgets, in which users can vote.
- in the application, delete your 2 votes for the previous budgets in test 3.
- as an admin, modify the budget's component used in test 3., and set Maximum number of budgets that user can vote on" to 1.
- complet the voting process for the one budget, and see that you have a modal with custom message n°2 as you have voted for the maximum available budgets.
- see also that you can't access to the second budget projects page.

5. Testing the grid/list design in projects index
- as a user, go to the projects index page
- see that the default view of the project is grid
- click on the list icon and see that the projects are in list mode 
- click on grid icon and see that you are back to grid view 

<img width="1028" alt="Capture d’écran 2024-12-27 à 10 22 23" src="https://github.com/user-attachments/assets/3c585e5c-fd51-4a86-a1cd-f389cff7a288" />


6. Testing redirection for a process with only one budget
- In the BO, go to a process with only one budget
-  Don't define a "URL to redirect user after voting on all available budgets",  but add a custom text in "Popup text after voting in all available budgets"
- in the FO, go to the budget, add complete your vote
- See that you are redirected to the process show page and that you have a modal with your custom text

7. Testing that vote button is not present when vote is finished
- in the BO, go to a process =>  configure budget component => select in the Step settings "Voting
finished"
- in the FO, go to the budget => projects and see that the button to vote for a project is not displayed in grid mode and in list mode

8. Testing that "Add to vote" button is not displayed when vote is disabled
- in the BO, go to a process =>  configure budget component => select in the Step settings "Voting
disabled"
- in the FO, go to the budget => projects and see that the button to vote for a project is not displayed in grid mode and in list mode, and that a text indicating "Vote disabled" is displayed on the left.

####Tasks
 [X] Update specs
 [X] Add specs
